### ISSUE TITLE
Ring enchant replace sims all permutations as opposed to all combinations

### ISSUE THREAD

**Issue by eamonno** [seriallos/raidbots-issues#116]
If you select the option to sim all four ring enchants you need to do 9 combinations:

crit-crit
crit-haste
crit-mastery
crit-vers
haste-haste
haste-mastery
haste-vers
mastery-mastery
mastery-vers
vers-vers

Instead the sim does all 16 permutations which includes a lot of redundant options (e.g. having your two rings enchanted mastery-haste is the same as having haste-mastery in the other order). When simming gem replacements the sim seems to work out all the combinations but ring enchants for some reason use all permutations (I don't have a fully geared dual wielder to check how weapon enchants works).

Functionally this gives the correct answer but it would be generating a lot of extra server load that could be avoided.

**Comment by seriallos** [comment]
Yeah, that's been on my list to look at for a while. The code to handle enchants is different from gems which is why gems don't have that problem.

Definitely want to look into it but don't have an ETA
